lib/internal/modules/esm/loader.js

```diff
@@ -64,6 +66,14 @@ class Loader {
     if (format === 'builtin')
       return { url: `node:${url}`, format };
 
+    if (this._resolve !== defaultResolve) {
+      try {
+        new URL(url);
+      } catch (err) {
+        throw new ERR_INVALID_URL(url);
+      }
```

Comment:
hmmm... I'm not sure what the benefit of the try/catch and additional throw here. If the `url` is invalid, `new URL(url)` will throw and accomplish the same thing, right?

Reply:
Well caught, yes that's simpler. Updated.